src/main/webapp/2Dscript.js

```diff
@@ -80,7 +175,418 @@ function addCenterControl(controlDiv) {
+      currentLat, priorLat, currentLng, priorLng);
+  // Choose a subSectionNumber, implement differently in future pr.
+  const subSectionNumber = 1;
+  // Clear prior paths, only display newly selected ones.
+  if (markerBottom !== undefined) {
+    markerBottom.setMap(null);
+  }
+  if (markerTop !== undefined) {
+    markerTop.setMap(null);
+  }
+  if (subPath !== undefined) {
+    subPath.setMap(null);
+  }
+  subPath = getPolyLine(subLine, 'blue', 1.0, 2);
+  // Setup event listener to show option for 3D window when polyline is
+  // clicked.
+  google.maps.event.addListener(subPath, 'click', linkTo3D);
+  subPath.setMap(map);
+  markerBottom = genMarker(subSectionData.minLatPoint.lat,
+      subSectionData.minLatPoint.lng);
+  markerTop = genMarker(subSectionData.maxLngPoint.lat,
+      subSectionData.maxLngPoint.lng);
+  markerBottom.setMap(map);
+  markerTop.setMap(map);
+
+  sessionStorage.setItem(id + '_' + type + '_' + subSectionNumber,
+      JSON.stringify(subLine));
+}
+
+/**
+ * Change the rectangle width to be match the current cursor position.
+ * @param {MouseEvent} event
+ */
+function genChangingBox(event) {
+  if (isMouseDown) {
+    console.log('Set width triggered');
+    finX = event.pageX;
+    finY = event.pageY;
+    widget.width(finX - x);
+    widget.height(finY - y);
+    widget.css({
+      'width': (finX - x) + '!important',
+      'height': (finY - y) + '!important',
+      'display': 'block',
+      'border': '2px dashed #ccc',
+    });
+  }
+}
+
+/**
+ * Generate a Google maps marker at the given lat and long.
+ * @param {number} latitude
+ * @param {number} longitude
+ * @return {google.maps.Marker}
+ */
+function genMarker(latitude, longitude) {
+  const marker = new google.maps.Marker({
+    position: {lat: latitude, lng: longitude},
+    title: 'Lat: ' + latitude + ' Lng: ' + longitude,
+  });
+  return marker;
+}
+
+
+/**
+ * Create a polyline from the given points with specified parameters.
+ * @param {Point} linePoints
+ * @param {string} color
+ * @param {number} opacity
+ * @param {number} weight
+ * @param {number} index
+ * @return {google.maps.Polyline}
+ */
+function getPolyLine(linePoints, color, opacity, weight, index = 1) {
+  const polyLine = new google.maps.Polyline({
+    path: linePoints,
+    geodesic: true,
+    strokeColor: color,
+    strokeOpacity: opacity,
+    strokeWeight: weight,
+    zIndex: index,
+  });
+  return polyLine;
+}
+/**
+ * @typedef {Object<Array<Point>, Point, Point>} subSectionData
+ * @property {Array<Point>} subLine
+ * @property {Point} minLatPoint
+ * @property {Point} maxLngPoint
+ */
+/**
+ * Compute the subsection of a line contained in a bounding rectangle.
+ * @param {Array<Point>} pose
+ * @param {number} currentLat
+ * @param {number} priorLat
+ * @param {number} currentLng
+ * @param {number} priorLng
+ * @return {subSectionData}
+ */
+function computeSubSection(pose, currentLat, priorLat, currentLng, priorLng) {
```

Comment:
I think `currentLat, currentLng, priorLat, priorLng` would be a better order. We're conditioned to see lat,lng pairs together.